packages/tizen_package_manager/tizen/src/tizen_package_manager_plugin.cc

```diff
@@ -218,20 +219,31 @@ class TizenPackageManagerPlugin : public flutter::Plugin {
   }
 
   void GetAllPackagesInfo(std::unique_ptr<FlMethodResult> result) {
-    TizenPackageManager &package_manager = TizenPackageManager::GetInstance();
-    std::optional<std::vector<PackageInfo>> packages =
-        package_manager.GetAllPackagesInfo();
-    if (!packages.has_value()) {
-      result->Error(std::to_string(package_manager.GetLastError()),
-                    package_manager.GetLastErrorString());
-      return;
-    }
-
-    flutter::EncodableList list;
-    for (const PackageInfo &package : packages.value()) {
-      list.push_back(flutter::EncodableValue(PackageInfoToMap(package)));
-    }
-    result->Success(flutter::EncodableValue(list));
+    // TizenPackageManager::GetAllPackagesInfo() is an expensive operation and
+    // might cause unresponsiveness on low-end devices if run on the platform
+    // thread.
+    ecore_thread_run(
+        [](void *data, Ecore_Thread *thread) {
+          auto *result = static_cast<FlMethodResult *>(data);
+
+          TizenPackageManager &package_manager =
+              TizenPackageManager::GetInstance();
+          std::optional<std::vector<PackageInfo>> packages =
+              package_manager.GetAllPackagesInfo();
```

Comment:
Would it be safe if the caller thread (main thread) exits before the operation finishes?

I don't think we need a lock here because the operation is read-only.